woocommerce/class-sv-wc-plugin.php

```diff
@@ -45,6 +45,9 @@ abstract class SV_WC_Plugin {
 	/** Plugin Framework Version */
 	const VERSION = '5.10.14';
 
+	/** Default support for HPOS */
+	const SUPPORTS_HPOS = true;
```

Comment:
I would use a protected property for this

Reply:
it would be in line with plugins setting arguments from their constructor - that's when they could set this internal flag

Reply:
the parent constructor can apply properties passed from args as it does currently

Reply:
I think we should put an explicit notice in the changelog warning developers to be sure to set this flag to `false` if they're using this version of the framework and _haven't_ yet implemented HPOS compatibility

Reply:
or perhaps we can be more conservative about it and default to `false` - but either way that could have unintended effects if the devs are not sure how to implement that... I'm actually leaning to default `false`

Reply:
@unfulvio-godaddy I agree with your opinion. we should use the `protected` property with a default value is `false`.

Reply:
Because constant always priests in memory.

Reply:
@unfulvio-godaddy My intention to make it constant is that somehow it helps us to detect whether the weather plugin supports HPOS.